core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -217,6 +219,9 @@ public void runInContext() {
       updateTimeoutHeaders(effectiveDeadline, callOptions.getDeadline(),
           context.getDeadline(), headers);
       ClientTransport transport = clientTransportProvider.get(callOptions, headers);
+      if (transport instanceof ConnectionClientTransport) {
```

Comment:
hmm, should we add `getAttrs` to `ClientTransport` to avoid cast here? I'm not sure that's a right change, but we would then kill `ConnectionClientTransport`?